### PR TITLE
feat: [P4PU-716] Cart icon & drawer

### PR DESCRIPTION
**Pull request by mmoio**
#### List of Changes

- Replaces mui-italia HeaderProduct
  with SubHeader component
- Fixes SubHeader logo Styling
- Adds cart icon to SubHeader
- Adds CartDrawer component
- Handles Cart app state through signals
- Adds localization
- Adds styles files to vitest excluded files
- unit tests

#### Motivation and Context

This pr adds the first basis of the cart functionality. Only drawer empty state is considered
#### How Has This Been Tested?
- Visual testing
- Unit tests

#### Screenshots (if appropriate):
image
image

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/Header/SubHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  const {
+    state: { cart }
+  } = useStore();
+
+  const { toggleCartDrawer } = useCartActions();
+
+  const Cart = () => {
+    return (
+      <Button variant="naked" sx={{ gap: 1 }} onClick={toggleCartDrawer} name="CartButton">
```
Comment (vito80ba) [inline src/components/Header/SubHeader.tsx:24]: Please add a proper title or aria-label to identify the action trigged by this button

File: src/components/Header/SubHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        <Typography variant="inherit">{cart.amount}</Typography>
+        <ShoppingCartIcon fontSize="small" />
+      </Button>
+    );
+  };
+
+  const Product = () => (
+    <Button href={ArcRoutes.DASHBOARD} size="medium" target="_self" sx={{ borderRadius: 2 }}>
```
Comment (vito80ba) [inline src/components/Header/SubHeader.tsx:32]: Also here, define an accessibility label

**Review by stratoivandiluccio — CHANGES_REQUESTED** [review]
Please, create a new long lived feature branch and use that branch as target of the PR. This is necessary because the cart functionality isn't included in the MVP

File: src/components/Header/SubHeader.tsx (hunk 1 above)
Comment (mmoio) [reply src/components/Header/SubHeader.tsx]: I've refactored this component to improve a11y in the latest commit

**Comment by mmoio** [comment]
> Please, create a new long lived feature branch and use that branch as target of the PR. This is necessary because the cart functionality isn't included in the MVP

I've created the branch [feat/P4PU-715](https://github.com/pagopa/arc-fe/tree/feat/P4PU-715-cart) and changed this pr target to it

**Review by vito80ba — APPROVED** [review]

**Pull request merged**